apis/v1alpha2/httproute_types.go

```diff
@@ -663,7 +686,43 @@ type HTTPRequestHeaderFilter struct {
 	Remove []string `json:"remove,omitempty"`
 }
 
-// HTTPRequestRedirectFilter defines configuration for the RequestRedirect filter.
+// HTTPPathModifierType defines the type of path redirect.
+type HTTPPathModifierType string
+
+const (
+	// This type of modifier indicates that the complete path will be replaced
+	// by the path redirect value.
+	AbsoluteHTTPPathModifier HTTPPathModifierType = "Absolute"
+
+	// This type of modifier indicates that any prefix path matches will be
+	// replaced by the substitution value. For example, a path with a prefix
+	// match of "/foo" and a ReplacePrefixMatch substitution of "/bar" will have
+	// the "/foo" prefix replaced with "/bar" in matching requests.
+	PrefixMatchHTTPPathModifier HTTPPathModifierType = "ReplacePrefixMatch"
+)
+
+// HTTPPathModifier defines configuration for path modifiers.
+// <gateway:experimental>
+type HTTPPathModifier struct {
+	// Type defines the type of path modifier.
+	//
+	// <gateway:experimental>
```

Comment:
Can we change the generator to automatically use the presence of this tag at a parent level to be propagated down?
Repeating this on each member seems error-prone to me.

I can't think of a situation where the struct is itself marked as experimental and any of the fields within would not be experimental, hence the suggestion.

Reply:
I agree, this would be really nice. But I don't know how we would ensure that the generated API docs had the tags then.

Reply:
Yeah it's actually unnecessary to have these tags on the child fields - the propagation @hbagdi describes already works when it comes to CRD generation. The problem is that it's not quite as clear when it comes to documentation, so I added this for _everything_ that was experimental at every level.